### PR TITLE
Add filter for recorded

### PR DESCRIPTION
**Pull request by imranq2**

### CODE REVIEW

File: src/operations/search/query/r4.js [hunk 1]
```diff
@@ -122,7 +122,23 @@ module.exports.buildR4SearchQuery = (resource_name, args) => {
         columns.add('meta.lastUpdated');
     }
 
-    for (const [field, filterObj] of Object.entries(customQueries)) {
+    for (const [resourceType, filterObj] of Object.entries(customScalarQueries)) {
```
Comment (DarylCheung) [inline src/operations/search/query/r4.js:125]: Is it possible to use the Array.filter() method to filter out the 'resourceType === resource_name' to help reduce nesting?

File: src/operations/search/query/r4.js [hunk 2]
```diff
@@ -122,7 +122,23 @@ module.exports.buildR4SearchQuery = (resource_name, args) => {
...
+        if (resourceType === resource_name) {
```
Comment (DarylCheung) [inline src/operations/search/query/r4.js:126]: Minor thing with the variable naming conventions of camel vs underscore.

File: src/operations/search/query/r4.js [hunk 3]
```diff
@@ -122,7 +122,23 @@ module.exports.buildR4SearchQuery = (resource_name, args) => {
...
+            for (const [property, propertyObj] of Object.entries(filterObj)) {
+                if (args[`${property}`]) {
+                    // console.log(property + ' = ' + propertyObj.type + ' , ' + propertyObj.field);
```
Comment (DarylCheung) [inline src/operations/search/query/r4.js:129]: Remove unneeded logging

File: src/operations/search/query/r4.js [hunk 4]
```diff
@@ -122,7 +122,23 @@ module.exports.buildR4SearchQuery = (resource_name, args) => {
...
+                    switch (propertyObj.type) {
+                        case 'instant':
```
Comment (DarylCheung) [inline src/operations/search/query/r4.js:131]: Will the switch case here grow? If it does, should we create a separate method for it to make it easier to test.

File: src/tests/queryFilters/auditEventsRecorded/auditEventsRecorded.test.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    beforeEach(async () => {
+        await commonBeforeEach();
+    });
+
+    afterEach(async () => {
+        await commonAfterEach();
+    });
+
+    describe('AuditEvent Recorded Tests', () => {
+        test('search by recorded works', async () => {
```
Comment (DarylCheung) [inline src/tests/queryFilters/auditEventsRecorded/auditEventsRecorded.test.js:23]: For test cases in the future, should we go with the AAA (arrange, act, assert) commenting?

File: src/operations/search/query/r4.js (hunk 1 above)
Comment (imranq2) [reply src/operations/search/query/r4.js:125]: this is unpacking so I'm not sure how we could filter...

File: src/operations/search/query/r4.js (hunk 2 above)
Comment (imranq2) [reply src/operations/search/query/r4.js:126]: fixed

File: src/operations/search/query/r4.js (hunk 3 above)
Comment (imranq2) [reply src/operations/search/query/r4.js:129]: fixed

File: src/operations/search/query/r4.js (hunk 4 above)
Comment (imranq2) [reply src/operations/search/query/r4.js:131]: we can refactor in the future when it grows

File: src/tests/queryFilters/auditEventsRecorded/auditEventsRecorded.test.js (hunk 5 above)
Comment (imranq2) [reply src/tests/queryFilters/auditEventsRecorded/auditEventsRecorded.test.js:23]: yes, we should start doing that

**Pull request merged**